### DISCUSSION TITLE
Best way to configure a custom preset for vehicles? And a couple of questions regarding Styles_TI_Lora

### DISCUSSION THREAD

**Discussion by Nobrumski** [AIrjen/OneButtonPrompt#174] [Q&A]
I am trying to create a custom preset to make various vehicles and not sure the best way to go about it. I was able to create a preset, but it seems extremely hit or miss on if anything included is actually a vehicle. For example I ended up with a Funko pop, and a Japanese landscape. 

This is what the Json looks like:
  "Vehicles": {
    "insanitylevel": 3,
    "subject": "object",
    "artist": "all",
    "chosensubjectsubtypeobject": "vehicles",
    "chosensubjectsubtypehumanoid": "all",
    "chosensubjectsubtypeconcept": "all",
    "chosengender": "all",
    "imagetype": "all",
    "imagemodechance": 100,
    "givensubject": "",
    "smartsubject": true,
    "givenoutfit": "",
    "prefixprompt": "",
    "suffixprompt": "",
    "giventypeofimage": "",
    "antistring": ""

I tried changing chosensubjectssubtypehumanoid to vehicles just since I routinely was seeing people. For example out of a 4 batch run,  one was a hovercraft, but the other three were, 2 people in a canoe, 1 woman in a canoe, 1 woman standing at the bridge of a starship.  Second run was better with a surfboard, a police car, a classic formula one car, and then a lava pit. 

Prompt:RAW photo, (Funicular:1.2) , lava pit, Amusing, Sony A7, L USM, Rule of Thirds, by artgerm, greg rutkowski and alphonse mucha, loish and wlop, dynamic dramatic atmosphere, striking, fine detail, ambient atmosphere, holy, highly decorated, rich vivid colors, sublime, great composition, unique

I also tried changing "giventypeofimage": "", to "vehicles" which seemed to help alot though I still get random portraits of people though it is possibly the checkpoint was confused on what a wind surfboard horse drawn carriage was lol. 
Prompt ([Wind surfboard:Horse-drawn carriage:5]:1.2) , detailed with details, emotive black and white portraits, glamorous, influenced by Native American art and culture, religious portraits, cinematic, stunning detail, luxury, intricate detail, dynamic cinematic perfect background, dynamic, deep aesthetic

I don't expect magic every time, but I feel I am missing a setting since some of the randomness is unrelated. 

On the Styles_TI_Lora
With Loras being incompatible, will I have any problems adding both SD 1.5 and SDXL and Pony Loras? New to this so not sure if it will just ignore the incompatible Lora or cause issues. I didn't want to go through adding all my Loras only to find out I should have done it differently.

I also wasn't sure if there was a way to restrict Loras to specific presets. I have some Loras which are specifically for vehicles for example so wouldn't go well in the Styles_ti_lora like [Truck Maker](url=https://civitai.com/models/94842) so would prefer it not show up in DND Portraits as an example.

**Comment by AIrjen** [comment]
Hi,

There are some things at work here. So the first one, is your main model. If it has a tendency to produce females, it will try to add them in. Try a more generalised model for vehicles, or one that is attuned to it.
For example, these is a Funicular result of your prompt on "leosamsHelloworldSDXL_helloworldSDXL40"

![2024-03-29_12-54-21_3796](https://github.com/AIrjen/OneButtonPrompt/assets/130234949/900705dc-268a-4b9b-8ae0-a3e92a886ecc)

Second, a lot of artists in the dataset of SD have a people focus and can overpower your intentions. You might get better consistent results by setting artists to "none".

Third, I tried a bit with these settings, you can try adding "object" in the prefix prompt. That helped me a little while playong around. You can also try "product shot"

Try this:

` "vehicles": {
    "insanitylevel": 5,
    "subject": "object",
    "artist": "none",
    "chosensubjectsubtypeobject": "vehicles",
    "chosensubjectsubtypehumanoid": "all",
    "chosensubjectsubtypeconcept": "all",
    "chosengender": "all",
    "imagetype": "all",
    "imagemodechance": 20,
    "givensubject": "",
    "smartsubject": true,
    "givenoutfit": "",
    "prefixprompt": "object",
    "suffixprompt": "",
    "giventypeofimage": "",
    "antistring": ""
  }
}`

**Comment by AIrjen** [comment]
For your other question, the standard Styles_TI_Lora is for stylings.

What you can do, is in the \userfiles\ directory, add a file called vehicles_addon.csv
In this csv, add a line:
`truck <lora:truck:1>`

Now it will sometimes pull the truck including LoRA when choosing a vehicle to generate.

Hope that helps!

**Comment by Nobrumski** [comment]
Thanks, that makes sense. I didn't even think of the checkpoint leaning more towards people. 

Definitely getting more along the lines of what I was looking to see. 
![00104-leosamsHelloworldXL_helloworldXL50GPT4V - 3630397362](https://github.com/AIrjen/OneButtonPrompt/assets/1368073/98aae84c-329a-4598-9dea-7cec79895149)
![00107-leosamsHelloworldXL_helloworldXL50GPT4V - 3630397365](https://github.com/AIrjen/OneButtonPrompt/assets/1368073/a2586d92-486d-408b-856d-fe0d055456c7)
